### PR TITLE
Improved Zally UX

### LINKED ISSUE #192: Resolve to a PSI element from Zally inspection
Zally provides a path to a key/value that caused the violation. We should use this path to link the Intellij error to a PSI element. This will improve the developer experience; a user can click the error/warning and the cursor is moved to the correct position.

### PR DESCRIPTION
**Pull request by semonte**
Link warnings to proper PSI elements.

image

Fixes #192

### CODE REVIEW

File: examples/extensions-zalando/src/main/java/org/zalando/intellij/swagger/examples/extensions/zalando/validator/zally/ZallyYamlFileValidator.java [hunk 1]
```diff
@@ -51,7 +53,7 @@ private LintingResponse lint(final PsiFile file) {
         }
     }
 
-    public boolean shouldLint(final PsiFile file) {
+    private boolean shouldLint(final PsiFile file) {
```
Comment (maxim-tschumak) [inline examples/extensions-zalando/src/main/java/org/zalando/intellij/swagger/examples/extensions/zalando/validator/zally/ZallyYamlFileValidator.java]: Maybe, sine the latest change, `canLint()` would be a better name?

File: examples/extensions-zalando/src/main/java/org/zalando/intellij/swagger/examples/extensions/zalando/validator/zally/ZallyYamlFileValidator.java [hunk 2]
```diff
@@ -62,36 +64,45 @@ private boolean hasZallyUrl() {
...
 
         return problems.toArray(ProblemDescriptor.EMPTY_ARRAY);
     }
 
-
+    private Optional<YAMLKeyValue> getPsiElement(final String jsonPointer, final PsiFile file) {
+        Optional<Optional<PsiElement>> psiElement = Optional.ofNullable(jsonPointer)
+                .map(pointer -> pointer
+                        .replace("/", ".")
+                        .replace("~1", "/"))
```
Comment (maxim-tschumak) [inline examples/extensions-zalando/src/main/java/org/zalando/intellij/swagger/examples/extensions/zalando/validator/zally/ZallyYamlFileValidator.java]: We should probably also include `~0` to make the mapping complete. ([related StackOverflow question/answer](https://stackoverflow.com/questions/31483170/purpose-of-tilde-in-json-pointer))

**Review by maxim-tschumak — APPROVED** [review]
:+1:

**Comment by semonte** [comment]
I have addressed the comments at https://github.com/zalando/intellij-swagger/pull/193/commits/07462bc72f0f6b02d608711a2aacfa0b71f61a25

**Comment by maxim-tschumak** [comment]
:+1:

**Pull request merged**